tools/docker/Dockerfile

```diff
@@ -1,33 +1,15 @@
-# Since we support arm64, take advantage of Docker and Zig's cross
-# compilation, rather than running under qemu.
-# https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
 FROM --platform=$BUILDPLATFORM ubuntu:22.04 AS build
 
-RUN  apt-get update \
-  && apt-get install -y wget xz-utils git
-
+# NB: We are doing something unconventional here. To ensure that docker images contain the exact
+# same TigreBeetle binary we publish as our release, we don't build inside a container, and merely
+# extract the right pre-exist binary (we pick among for options based on build-time DEBUG and
+# TARGETPLATFORM arguments).
+RUN  apt-get update && apt-get install -y unzip
 WORKDIR /opt/beta-beetle
-
-# Copy install_zig by itself, to try and keep cache hits as high as possible.
-RUN mkdir scripts
-COPY scripts/install_zig.sh ./scripts
-
-ENV PATH="${PATH}:/opt/beta-beetle/zig"
-
-# Allows passing in --build-arg DEBUG=true during builds (needed by Github CI for generating debug image)
+COPY ./dist/tigerbeetle .
 ARG DEBUG=false
-
-# With our multiplatform builds, Docker will only repeat steps after the ARG TARGETPLATFORM.
-# So, explicitly install zig before then, to not need to do it twice.
-RUN ./scripts/install_zig.sh
-
-COPY .git .git
-COPY scripts ./scripts
-COPY src ./src
-COPY build.zig ./build.zig
-
 ARG TARGETPLATFORM
-RUN test ${TARGETPLATFORM} = "linux/arm64" && CPU="baseline+aes+neon" || CPU="x86_64_v3+aes"; CI=true ./scripts/install.sh
+RUN unzip tigerbeetle-$(test ${TARGETPLATFORM} = "linux/arm64" && echo "aarch64" || echo "x86_64")-linux$(test ${DEBUG} = "true" && echo "-debug" || echo "").zip
```

Comment:
Ah, either if doing my suggestion below or with the current setup, I don't think `tigerbeetle` will be `chmod +x`'d anywhere (IIRC zip doesn't keep permission...)

Reply:
I think it gets the `x` bit actually, at least locally `docker run` seems to work sometimes (it seems podman picks a random arch from a multiarch builds and sometises you get lucky)